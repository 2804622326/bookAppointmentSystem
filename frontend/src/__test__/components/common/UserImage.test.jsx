import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserImage from '../../../components/common/UserImage.jsx';

test('renders placeholder when no photo and snapshot', () => {
  const { asFragment } = render(<UserImage userId={1} userPhoto={null} />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders user photo when provided', () => {
  const base64 = btoa('img');
  const { asFragment } = render(<UserImage userId={1} userPhoto={base64} />);
  expect(asFragment()).toMatchSnapshot();
});
