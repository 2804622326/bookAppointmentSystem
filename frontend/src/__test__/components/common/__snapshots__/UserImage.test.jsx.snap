// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders placeholder when no photo and snapshot 1`] = `
<DocumentFragment>
  <img
    alt="User photo"
    class="card-img user-image"
    src="test-file-stub"
  />
</DocumentFragment>
`;

exports[`renders user photo when provided 1`] = `
<DocumentFragment>
  <img
    alt="User photo"
    class="card-img user-image"
    src="data:image/png;base64,aW1n"
  />
</DocumentFragment>
`;
